### PR TITLE
License audit

### PR DESCRIPTION
**Pull request by lionellloh**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by lionellloh** [comment]
- Added Audit for license DSL for formula. 
- Audit checks if a license info is a verified spdx-id
- If `--online` flag is specified, and has github info, audit will do a comparison check to make sure what the formulae states matches what github API detects 
- Wrote tests to verify the behaviour for above

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -109,7 +110,9 @@ def audit
 
     # Check style in a single batch run up front for performance
     style_results = Style.check_style_json(style_files, options) if style_files
-
+    # load licenses
+    path = File.join(File.dirname(__FILE__), "spdx.json")
+    spdx_ids = JSON.parse(File.open(File.expand_path(path)).read)
```
Comment (whoiswillma) [inline Library/Homebrew/dev-cmd/audit.rb:115]: Is it alright if the file is never explicitly closed?

File: Library/Homebrew/test/dev-cmd/audit_spec.rb [hunk 2]
```diff
@@ -95,6 +95,83 @@ class Foo < Formula
       end
     end
 
+    describe "#audit_license" do
+      let(:spdx_ids) {
+        full_path = File.join(File.expand_path(File.dirname(__FILE__) + "../../../dev-cmd/"), "spdx.json")
```
Comment (whoiswillma) [inline Library/Homebrew/test/dev-cmd/audit_spec.rb:100]: Could we use `File.join` to join `File.dirname(__FILE__)` and `../../../dev-cmd`?

File: Library/Homebrew/test/dev-cmd/audit_spec.rb [hunk 3]
```diff
@@ -95,6 +95,83 @@ class Foo < Formula
...
+
+        fa.audit_license
+        expect(fa.problems).to be_empty
+      end
+
+      it "checks online and verifies that a standard license id is the same "\
+        "as what is indicated on its Github repo" do
+        fa = formula_auditor "cask", <<~RUBY, spdx_ids: spdx_ids, online: true, core_tap: true
+          class Cask < Formula
+              url "https://github.com/cask/cask/archive/v0.8.4.tar.gz"
```
Comment (whoiswillma) [inline Library/Homebrew/test/dev-cmd/audit_spec.rb:148]: nit: 4 spaces indentation here, should it be 2?

File: Library/Homebrew/test/dev-cmd/audit_spec.rb [hunk 4]
```diff
@@ -95,6 +95,83 @@ class Foo < Formula
...
+          end
+        RUBY
+
+        fa.audit_license
+        expect(fa.problems).to be_empty
+      end
+
+      it "checks online and detects that a formula-specified license is not "\
+        "the same as what is indicated on its Github repository" do
+        # odie "lol"
```
Comment (whoiswillma) [inline Library/Homebrew/test/dev-cmd/audit_spec.rb:160]: Delete

File: Library/Homebrew/dev-cmd/audit.rb (hunk 1 above)
Comment (lionellloh) [reply Library/Homebrew/dev-cmd/audit.rb:115]: Ok, using a block to close it automatically

**Pull request merged**